walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,92 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.PriorityQueue;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return new ArrayList<>();
+    } 
+    
+    PriorityQueue<TimeRange> busyTimeRanges = getBusyTimeRanges(events, request);
+    ArrayList<TimeRange> combinedBusyTimeRanges = getCombinedBusyTimeRanges(busyTimeRanges);
+    Collection<TimeRange> result = getAvailableTimeRanges(combinedBusyTimeRanges, request);
+    return result;
+  }
+
+  private boolean eventHasRequestAttendees(Collection<String> requestAttendees, Event event) {
+    Set<String> eventAttendees = event.getAttendees();
+
+    for (String requestAttendee : requestAttendees) {
```

Comment:
do you think we can modify this to one liner using
`Collections.disjoint()`?

Reply:
Thanks for the suggestion! Did not know about that :O